lib/puppet_x/cisco/aci_provider_utils.rb

```diff
@@ -199,112 +200,111 @@ def self.set_property_flush_attrs_frm_resource(p_inst)
           case provider_inst.property_flush[:ensure]
           when :present
             resp = Puppet::Util::NetworkDevice.current.post(
-                    self.gen_url4instance(provider_inst.property_flush[:dn]),
-                    self.gen_json4createmodify(provider_inst))
-            resp["imdata"].each do |mo|
-              resp_attrs_hash = self.xtract_attr_hash_frm_json(mo,
+              gen_url4instance(provider_inst.property_flush[:dn]),
+              gen_json4createmodify(provider_inst))
+            resp['imdata'].each do |mo|
+              resp_attrs_hash = xtract_attr_hash_frm_json(mo,
                                                           provider_inst.class)
-              self.debug_what_changed(resp_attrs_hash,
-                                      provider_inst.property_hash)
+              debug_what_changed(resp_attrs_hash,
+                                 provider_inst.property_hash)
             end
           when :absent
-            resp = Puppet::Util::NetworkDevice.current.delete(
-                      self.gen_url4instance(provider_inst.property_flush[:dn]))
-          end
-          #Return the corresponding APIC GUI URI that will be used for informational
-          #message
-          self.gen_url4apicgui(provider_inst.property_flush[:fvtenant],
-            provider_inst.property_flush[:dn])
-        rescue Exception => e
-            puts e.message
-            puts e.backtrace.inspect
+            Puppet::Util::NetworkDevice.current.delete(
+              gen_url4instance(provider_inst.property_flush[:dn]))
           end
+          # Return the corresponding APIC GUI URI that will be used for informational
+          # message
+          gen_url4apicgui(provider_inst.property_flush[:fvtenant],
+                          provider_inst.property_flush[:dn])
+        rescue => e
+          puts e.message
+          puts e.backtrace.inspect
+        end
       end
 
       #=============ACI Factor related utility methods =======================
 
-      #Extract Fabric Members from  DHCP client query request.
-      #@param [JSON] Response from the DHCP Client query
-      #@return [Hash] {NodeID => Attribute Map} for leaf and spine members
+      # Extract Fabric Members from  DHCP client query request.
+      # @param [JSON] Response from the DHCP Client query
+      # @return [Hash] {NodeID => Attribute Map} for leaf and spine members
       #               Please note that there is a possibility that this method
       #               can return a nil
       #
       def self.xtract_fabric_member_frm_dhcp_query(one_dhcp_clnt_json)
-        if (one_dhcp_clnt_json and
-          (values = one_dhcp_clnt_json['dhcpClient']) and
-          (attrs  = values['attributes']) and
-          (id = attrs['id']) and
-          ((attrs['nodeRole'] == 'leaf') or (attrs['nodeRole'] == 'spine')))
-          #create an NodeID => Node Details Hash
-          {id=>attrs}
+        if one_dhcp_clnt_json &&
```

Comment:
Same question here.. we used to assign values and id but not using?

Reply:
Same as above, there was a bug here but now fixed